### ISSUE TITLE
Ionic2 Modal early display prevents the host page to be displayed

### ISSUE THREAD

**Issue by architruc** [driftyco/ionic#6933]
#### Short description of the problem:

When a page immediately displays a modal on load (for example in ngAfterViewInit, ngOnInit...), then the modal is dismissed, the page remains hidden (The `<ion-page>` element is still missing the `show-page` CSS class).

#### What behavior are you expecting?

When the modal is dismissed, the page is normally displayed.

**Steps to reproduce:**
1. In a page `ngAfterViewInit` lifecycle hook, create and present a modal
2. Dismiss the modal (for example with a simple `dismiss` button in the modal)
3. The page is blank. Inspecting the DOM reveals that the `<ion-page>` tag is missing the `show-page` CSS class.

**Example of usecase:** requires authentication before continuing using the app. A modal is convenient to display from anywhere without losing eventual state of a work in progress.

From the plunker below:

**HomePage (hosting the modal):**

```ts
	ngAfterViewInit() {
		Page1.show(this.nav);
	}
```

**Page1 (the modal):**

```ts
	static show(nav:NavController):void {
		let loginPage = Modal.create(Page1);
		nav.present(loginPage);
	}
	
	dismissModal() { // called by the dismiss button
		this.viewCtrl.dismiss();
	}
```
 

**Which Ionic Version?** 2.x

#### Plunker that shows an example of your issue

http://plnkr.co/edit/Wjfr5Olzo8PCMGm1JZgW?p=preview

**Run `ionic info` from terminal/cmd prompt:** (paste output below)

Cordova CLI: 6.1.1
Gulp version:  CLI version 3.9.1
Gulp local:   Local version 3.9.1
Ionic Framework Version: 2.0.0-beta.8
Ionic CLI Version: 2.0.0-beta.25
Ionic App Lib Version: 2.0.0-beta.15
OS: Windows 8.1
Node Version: v5.11.1

**Comment by jgw96** [comment]
Hello! Thanks for opening an issue with us! I have created a plunker [here](http://plnkr.co/edit/qK8tFFbEBXe8C6KqOWvn?p=preview) that shows the proper way to open a modal when a page is loaded. Thanks for using Ionic!

**Comment by architruc** [comment]
Thanks for this example, it works well.

Personally, I still see it as a workaround, not very natural, since the page load is interrupted by the modal (at least kept hidden even after dismiss) if the modal is displayed too early (it is a kind of popup, not a new page), and I don't remember it is an expected behavior described in the documentation.

The [sample for ionic2/angularfire2](https://github.com/aaronksaunders/ionic2-angularfire-sample) suffers from it as well, since authentication is done in an early modal.

But that's just my opinion :)

**Comment by vidakovic** [comment]
@jgw96 this workaround is not working for me on an Android phone (using crosswalk-project). When I test it in a desktop browser the modal dialogs seem to work.

I tried multiple approaches:

1. sending the the dialog object to the main app component where NavController is injected via @ViewCtrl
2. inject NavController in the page constructors
3. inject App in the page constructors and use app.getActiveNav().present(...)

On Android I see - sometimes - a short flickering, i. e. the dialog shows for a split second, but disappears then.

I tried to debug the Android app with the Chrome debugger... sometimes (I think when I hold execution at a breakpoint a longer) then the dialog appears... and funny enough after that it seems to working as expected.

Any ideas?
